*** cuml/cpp/cmake/thirdparty/get_treelite.cmake-dist	2023-01-22 22:39:51.827639326 +0900
--- cuml/cpp/cmake/thirdparty/get_treelite.cmake	2023-01-22 22:40:31.491525611 +0900
***************
*** 90,96 ****
      rapids_export_find_package_root(BUILD Treelite [=[${CMAKE_CURRENT_LIST_DIR}]=] cuml-exports)
  endfunction()
  
! find_and_configure_treelite(VERSION     3.0.1
!                         PINNED_TAG  cb09d539dce7e67f60cc33e4fac6b95b7185847a
                          EXCLUDE_FROM_ALL  ${CUML_EXCLUDE_TREELITE_FROM_ALL}
                          BUILD_STATIC_LIBS ${CUML_USE_TREELITE_STATIC})
--- 90,96 ----
      rapids_export_find_package_root(BUILD Treelite [=[${CMAKE_CURRENT_LIST_DIR}]=] cuml-exports)
  endfunction()
  
! find_and_configure_treelite(VERSION     3.1.0
!                         PINNED_TAG  4103ed090f17ad16596c3b46357b810a5ccc3969
                          EXCLUDE_FROM_ALL  ${CUML_EXCLUDE_TREELITE_FROM_ALL}
                          BUILD_STATIC_LIBS ${CUML_USE_TREELITE_STATIC})
