*** cusignal/build.sh-dist	2022-12-05 15:35:12.833770311 +0900
--- cusignal/build.sh	2022-12-05 15:36:59.890133906 +0900
***************
*** 133,152 ****
          --generate-code arch=compute_70,code=sm_70 \
          --generate-code arch=compute_72,code=sm_72"
      
!     if [ "$NVCC_MAJOR" -lt 11 ]; then
!         GPU_ARCH="${GPU_ARCH} --generate-code arch=compute_75,code=[sm_75,compute_75]"
!         echo -e "\t including: CUDA 10.X - {50,52,53,60,61,62,70,72,75}"
!     else
!         GPU_ARCH="${GPU_ARCH} --generate-code arch=compute_75,code=sm_75"
!         if [ "$NVCC_MINOR" -eq 0 ]; then
!             GPU_ARCH="${GPU_ARCH} --generate-code arch=compute_80,code=[sm_80,compute_80]"
!             echo -e "\t including: CUDA 11.0 - {50,52,53,60,61,62,70,72,75,80}"
!         else
!             GPU_ARCH="${GPU_ARCH} --generate-code arch=compute_80,code=sm_80 \
                  --generate-code arch=compute_86,code=[sm_86,compute_86]"
!             echo -e "\t including: CUDA 11.1+ - {50,52,53,60,61,62,70,72,75,80,86}"
!         fi
!     fi
  }
  
  if (( ${BUILD_ALL_GPU_ARCH} == 0 )); then
--- 133,141 ----
          --generate-code arch=compute_70,code=sm_70 \
          --generate-code arch=compute_72,code=sm_72"
      
!      GPU_ARCH=" \
                  --generate-code arch=compute_86,code=[sm_86,compute_86]"
!      echo -e "\t including: CUDA 11.1+ - {50,52,53,60,61,62,70,72,75,80,86}"
  }
  
  if (( ${BUILD_ALL_GPU_ARCH} == 0 )); then
