*** cpp/build/_deps/libcudacxx-src/include/cuda/std/detail/__config-dist	2022-12-30 20:39:06.320176761 +0900
--- cpp/build/_deps/libcudacxx-src/include/cuda/std/detail/__config	2022-12-30 20:40:42.827943834 +0900
***************
*** 34,42 ****
      #ifndef __device__
          #define __device__
      #endif
!     #ifndef __forceinline__
!         #define __forceinline__
!     #endif
  #endif
  
  #ifdef _MSC_VER
--- 34,42 ----
      #ifndef __device__
          #define __device__
      #endif
! //    #ifndef __forceinline__
! //        #define __forceinline__
! //    #endif
  #endif
  
  #ifdef _MSC_VER
